### PR TITLE
don't allow any fields to be updated if template has task_plans

### PR DESCRIPTION
**Pull request by jivey**
As requested 1048, don't allow any fields to be changed if there are task_plans.

### CODE REVIEW

**Comment by Dantemss** [comment]
Hmm this is the kind of change I'd rather leave in the FE since it's not really necessary in the BE, but I'd not be opposed to merge if you added a comment saying that blocking all updates is not really 100% necessary, just our policy on grading template updates (so we remember in 6 months when they change their minds :).

**Comment by nathanstitt** [comment]
Yeah seems to me that this is a low stakes validation that could be handled in the FE.  I see the grading_templates api has a `has_task_plans` property so we should be able to just disable editing when that's true.

**Comment by Dantemss** [comment]
@nathanstitt we might have to modify it or add a new one that only counts published and open task plans to meet the requirement

**Comment by jivey** [comment]
I like the idea of enforcing this on the backend because it's supposed to check if the assignment is open + it affects students.

**Comment by nathanstitt** [comment]
yeah, that's true. I'm fine with it

File: app/subsystems/tasks/models/grading_template.rb [hunk 1]
```diff
@@ -76,6 +76,10 @@ def has_task_plans?
     !task_plans.reject(&:withdrawn?).empty?
   end
 
+  def has_open_task_plans?
+    task_plans.reject(&:withdrawn?).any?(&:out_to_students?)
+  end
```
Comment (jivey) [inline app/subsystems/tasks/models/grading_template.rb:81]: @Dantemss Does this cover the cases we want?

**Comment by jivey** [comment]
(as an aside, it would also be good to check it on the FE and show an error message when clicking Edit instead of opening the form, cause the experience with this approach isn't ideal.)

File: app/subsystems/tasks/models/grading_template.rb (hunk 1 above)
Comment (Dantemss) [reply app/subsystems/tasks/models/grading_template.rb:81]: I believe this is correct.

**Pull request merged**